js.go

```diff
@@ -806,20 +812,23 @@ func (js *js) handleAsyncReply(m *Msg) {
 		js.mu.Unlock()
 		return
 	}
-	// Remove
-	delete(js.pafs, id)
 
-	// Check on anyone stalled and waiting.
-	if js.stc != nil && len(js.pafs) < js.opts.maxpa {
-		close(js.stc)
-		js.stc = nil
+	closeStc := func() {
+		// Check on anyone stalled and waiting.
+		if js.stc != nil && len(js.pafs) < js.opts.maxpa {
+			close(js.stc)
+			js.stc = nil
+		}
 	}
-	// Check on anyone one waiting on done status.
-	if js.dch != nil && len(js.pafs) == 0 {
-		dch := js.dch
-		js.dch = nil
-		// Defer here so error is processed and can be checked.
-		defer close(dch)
+
+	closeDch := func() {
+		// Check on anyone one waiting on done status.
+		if js.dch != nil && len(js.pafs) == 0 {
+			dch := js.dch
+			js.dch = nil
+			// Defer here so error is processed and can be checked.
+			defer close(dch)
+		}
 	}
 
 	doErr := func(err error) {
```

Comment:
We should clear the reply subject before passing the message in error handler. If user wants to retry manually inside the error handler, having `paf.msg.Reply` set will cause the retry attempt to fail with `nats: reply subject should be empty`